ethcore/src/client/client.rs

```diff
@@ -1671,7 +1671,24 @@ impl BlockChainClient for Client {
 				};
 
 				let traces = self.tracedb.read().filter(&filter);
-				Some(traces)
+				if traces.is_empty() {
+					return None;
+				}
+
+				let mut traces_iter = traces.into_iter();
+				if let Some(after) = after {
+					if let TransactionId::Location(BlockId::Hash(block_hash), transaction_number) = after.transaction {
```

Comment:
Why does it quietly ignores `after` argument if it is `Some(TransactionId::Hash(_))`? I can see that in RPC only `TransactionId::Location` is used, but it is not good to have public method in separate (not RPC crate) which accepts two options and quietly works for single option only.